### ISSUE TITLE
Enhancement Request:  Ability to add container metadata to non-console output

### ISSUE THREAD

**Issue by PhaedrusTheGreek** [elastic/beats#6393]
Kubernetes and Docker metadata can be added to log lines shipped from container STDOUT automatically, and this works great when only one application is logging to STDOUT.  

As much as it is considered bad practice, sometimes we have scenarios where containers might have more than one application running inside for legacy reasons or otherwise.  Using [various hacks](https://serverfault.com/questions/599103/make-a-docker-application-write-to-stdout), it's possible to send output from multiple applications (log files, etc) to STDOUT of the container , but there is no way to differentiate the log lines.  All say `stream: "stdout"`, which I suppose is accurate, and an enhancement would need to be addressed in docker for an extension.

Currently there's 2 possibilities to handle this with Filebeat:
- Have the background application log to STDOUT via the hack listed above
 - GOOD: Container metadata is attached
 - BAD: no way to know which app sourced the message
- Have the background application log to a shared filesystem and have a Filebeat prospector process the logs from there
 - GOOD: We can differentiate the application
 - BAD: We lose the container metadata 

In Filebeat, is there anything we can do to preserve or attach container metadata to files harvested from shared filesystems?

**Comment by exekias** [comment]
Hi,

Both solutions are good, depending on your case, if you have several apps in the same container, you may need them to put some info in the log to differentiate them.

As for the shared filesystem, you can hardcode the container id (instead of using source file match). I just tested this config and it works:

```
- type: log
  enabled: true
  paths:
    - foo.log

  fields:
    container_id: dfbccab95a73

processors:
  - add_docker_metadata:
      match_short_id: true
      match_source: false
      match_fields: ["fields.container_id"]
```

**Comment by PhaedrusTheGreek** [comment]
Great workaround here.

Note that `match_short_id` is only available in 6.3+, so in prior releases you will have to omit this option and specify the full `container_id` (which can be grabbed by `docker ps --no-trunc`)
